rust-symcrypt/src/ecdh.rs

```diff
@@ -0,0 +1,270 @@
+/* EcDh functions. For further documentation please refer to symcrypt.h */
+
+use crate::curve_type::*;
+use crate::errors::SymCryptError;
+use std::vec;
+use symcrypt_sys;
+
+/// EcDhKey is a wrapper around symcrypt_sys::PSYMCRYPT_ECKEY, this is to let rust handle the free'ing of this pointer
+/// EcDhKey must be allocated after EcDhExpandedCurve and free'd before EcDhExpandedCurve is free'd
+struct EcDhKey {
+    inner: symcrypt_sys::PSYMCRYPT_ECKEY,
```

Comment:
```suggestion
    inner: NonNull<symcrypt_sys::SYMCRYPT_ECKEY>,
```

You guarantee by construction that this pointer is never null, so you can put that in the type system if you want. It doesn't make a huge difference since you don't have any null checks on this anyway though.